src/nvim/memline.c

```diff
@@ -1872,13 +1878,13 @@ int gchar_pos(pos_T *pos)
 ///
 /// @return  a pointer to a line in a specific buffer
 static char *ml_get_buf_impl(buf_T *buf, linenr_T lnum, bool will_change)
-  FUNC_ATTR_NONNULL_ALL
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_NONNULL_RET
```

Comment:
This was in search of failing assertion and test failures (before I realized `ml_line_textlen = 0` is the problem).